// electron-store/index.js

const fs = require('fs');
const path = require('path');

class ElectronStore {
  constructor(options = {}) {
    this.name = options.name || 'config';
    this.cwd = options.cwd || (process.env.HOME || '/home/chat2api/.chat2api');
    this.filePath = path.join(this.cwd, this.name + '.json');
    this.data = {};
    try {
      if (fs.existsSync(this.filePath)) {
        this.data = JSON.parse(fs.readFileSync(this.filePath, 'utf-8'));
      }
    } catch (e) {
      this.data = {};
    }
  }

  get(key, defaultValue) {
    const keys = key.split('.');
    let val = this.data;
    for (const k of keys) {
      if (val === null || val === undefined || typeof val !== 'object') return defaultValue;
      val = val[k];
    }
    return val !== undefined ? val : defaultValue;
  }

  set(key, value) {
    const keys = key.split('.');
    let obj = this.data;
    for (let i = 0; i < keys.length - 1; i++) {
      if (!obj[keys[i]] || typeof obj[keys[i]] !== 'object') obj[keys[i]] = {};
      obj = obj[keys[i]];
    }
    obj[keys[keys.length - 1]] = value;
    this._save();
  }

  delete(key) {
    const keys = key.split('.');
    let obj = this.data;
    for (let i = 0; i < keys.length - 1; i++) {
      if (!obj[keys[i]] || typeof obj[keys[i]] !== 'object') return;
      obj = obj[keys[i]];
    }
    delete obj[keys[keys.length - 1]];
    this._save();
  }

  clear() {
    this.data = {};
    this._save();
  }

  _save() {
    try {
      const dir = path.dirname(this.filePath);
      if (!fs.existsSync(dir)) fs.mkdirSync(dir, { recursive: true });
      fs.writeFileSync(this.filePath, JSON.stringify(this.data, null, 2));
    } catch (e) {
      console.error('[electron-store] save error:', e.message);
    }
  }
}

module.exports = ElectronStore;
