// electron/index.js
const path = require('path');
const os = require('os');

const app = {
  getPath: (name) => {
    const home = process.env.HOME || os.homedir();
    const dataDir = path.join(home, '.chat2api');
    switch (name) {
      case 'userData': return dataDir;
      case 'appData': return dataDir;
      case 'home': return home;
      case 'desktop': return path.join(home, 'Desktop');
      case 'documents': return path.join(home, 'Documents');
      case 'downloads': return path.join(home, 'Downloads');
      default: return path.join(dataDir, name);
    }
  },
  getVersion: () => '0.0.0',
  getName: () => 'chat2api',
  quit: () => process.exit(0),
  on: () => {},
  whenReady: () => Promise.resolve(),
  isPackaged: true,
};

const remote = { app };
const BrowserWindow = function() {};
BrowserWindow.prototype = { loadURL: () => {}, loadFile: () => {}, on: () => {} };

module.exports = {
  app,
  remote,
  BrowserWindow,
  ipcMain: { on: () => {}, handle: () => {} },
  ipcRenderer: { on: () => {}, send: () => {}, invoke: () => {} },
  shell: { openExternal: () => {} },
  dialog: { showSaveDialog: () => Promise.resolve({}), showOpenDialog: () => Promise.resolve({}) },
  Menu: { buildFromTemplate: () => ({ popup: () => {} }) },
  nativeImage: { createFromPath: () => ({ resize: () => {}, toPNG: () => Buffer.alloc(0) }) },
  contextBridge: { exposeInMainWorld: () => {} },
};
